cmd/endpoint.go

```diff
@@ -248,15 +254,64 @@ func checkCrossDeviceMounts(endpoints EndpointList) (err error) {
 	return mountinfo.CheckCrossDevice(absPaths)
 }
 
+func getPacks(packsFile string) ([][]string, error) {
+	r, err := os.Open(packsFile)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+
+	var slotArgs [][]string
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		slotArgs = append(slotArgs, strings.Fields(scanner.Text()))
+	}
+	prevSlotSize := len(slotArgs[0])
+	for i, args := range slotArgs {
+		if len(args) != prevSlotSize {
+			return nil, fmt.Errorf("Unexpected slot sizes found, slot %d is of length %d but expected %d", i+1, len(args), prevSlotSize)
+		}
+	}
+	return slotArgs, nil
+}
+
+// CreateEndpointsFromPacks - validates and creates new endpoints for given packs file.
+func CreateEndpointsFromPacks(serverAddr, packsFile string) (string, EndpointList, SetupType, int, error) {
+	var slotEndpoints EndpointList
+	var setupType SetupType
+
+	slotArgs, err := getPacks(packsFile)
+	if err != nil {
+		return serverAddr, nil, setupType, 0, err
+	}
+
+	var endpoints EndpointList
+	var slotSize int
+	for i, packs := range slotArgs {
+		serverAddr, endpoints, setupType, _, err = CreateEndpoints(serverAddr, packs...)
+		if err != nil {
+			return serverAddr, slotEndpoints, setupType, 0, err
+		}
+		var newEndpoints EndpointList
+		for _, endpoint := range endpoints {
+			endpoint.SlotIndex = i
+			newEndpoints = append(newEndpoints, endpoint)
+		}
+		slotSize = i + 1
+		slotEndpoints = append(slotEndpoints, newEndpoints...)
+	}
+	return serverAddr, slotEndpoints, setupType, slotSize, nil
+}
+
 // CreateEndpoints - validates and creates new endpoints for given args.
-func CreateEndpoints(serverAddr string, args ...string) (string, EndpointList, SetupType, error) {
+func CreateEndpoints(serverAddr string, args ...string) (string, EndpointList, SetupType, int, error) {
```

Comment:
Use `bool` instead of `int` ? (we are just returning either 0 or 1).
(And is it not possible to design away altogether? Line 291 above ignores the value.)